src/Controller/Api/MailArchiveController.php

```diff
@@ -0,0 +1,169 @@
+<?php declare(strict_types=1);
+
+namespace Frosh\MailArchive\Controller\Api;
+
+use Frosh\MailArchive\Content\MailArchive\MailArchiveEntity;
+use Frosh\MailArchive\Services\EmlFileManager;
+use Frosh\MailArchive\Services\MailSender;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\PlatformRequest;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Routing\Annotation\Route;
+use ZBateson\MailMimeParser\Header\AddressHeader;
+use ZBateson\MailMimeParser\Header\DateHeader;
+use ZBateson\MailMimeParser\Header\IHeader;
+use ZBateson\MailMimeParser\Header\Part\AddressPart;
+
+class MailArchiveController extends AbstractController
```

Comment:
We should avoid using AbstractController as a base class for controllers. We don't need something special from it.